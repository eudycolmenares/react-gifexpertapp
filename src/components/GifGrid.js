import PropTypes from 'prop-types'
import GifGridItem from "./GifGridItem"
import { useFetchGifs } from "../hooks/useFetchGifs"
// 
const GifGrid = ({ category }) => {    
    const { data: images, loading } = useFetchGifs( category )
    return (
        <>
            <h4 className="uppercase">{ category }</h4>
            { loading && <h5 className="animate__animated animate__flash animate__infinite">Loading...</h5> }
            <div className="card-grid">
                {
                    images.map( image =>
                        <GifGridItem 
                            key={ image.id }
                            { ...image } 
                        />
                    )
                }
            </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
// 
export default GifGrid