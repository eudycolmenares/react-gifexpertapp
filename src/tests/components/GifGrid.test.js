import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')
// 
describe('Tests in <GifGrid />', () => {
    const category = 'Avengers'
    test('Should display correctly!', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category={ category } /> )
        expect( wrapper ).toMatchSnapshot()
    })
    test('Shoud display item when images are obtained wwith useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything/thing.jpg',
            title: 'Anything'
        },{
            id: 'ABC2',
            url: 'https://localhost/anything/thing.jpg',
            title: 'Anything'
        },{
            id: 'ABC3',
            url: 'https://localhost/anything/thing.jpg',
            title: 'Anything'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow( <GifGrid category={ category } /> )
        // 
        // expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('h5').exists() ).toBe( false )
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
        
    })
})