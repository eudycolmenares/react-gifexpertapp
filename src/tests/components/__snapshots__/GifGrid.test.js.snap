// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> Should display correctly! 1`] = `
<Fragment>
  <h4
    className="uppercase"
  >
    Avengers
  </h4>
  <h5
    className="animate__animated animate__flash animate__infinite"
  >
    Loading...
  </h5>
  <div
    className="card-grid"
  />
</Fragment>
`;
